flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -237,15 +250,20 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
     // For step#4, we don't need to commit iceberg table again because in step#3 we've committed all
     // the files,
     // Besides, we need to maintain the max-committed-checkpoint-id to be increasing.
-    if (checkpointId > maxCommittedCheckpointId) {
-      LOG.info("Checkpoint {} completed. Attempting commit.", checkpointId);
-      commitUpToCheckpoint(dataFilesPerCheckpoint, flinkJobId, operatorUniqueId, checkpointId);
-      this.maxCommittedCheckpointId = checkpointId;
-    } else {
-      LOG.info(
-          "Skipping committing checkpoint {}. {} is already committed.",
-          checkpointId,
-          maxCommittedCheckpointId);
+    try {
+      if (checkpointId > maxCommittedCheckpointId) {
+        LOG.info("Checkpoint {} completed. Attempting commit.", checkpointId);
+        commitUpToCheckpoint(dataFilesPerCheckpoint, flinkJobId, operatorUniqueId, checkpointId);
+        this.maxCommittedCheckpointId = checkpointId;
+      } else {
+        LOG.info(
+            "Skipping committing checkpoint {}. {} is already committed.",
+            checkpointId,
+            maxCommittedCheckpointId);
+      }
+    } finally {
+      // reload the table in case new configuration is needed
+      this.table = tableSupplier.get();
```

Comment:
Shall we update the `this.manifestOutputFileFactory.manifestOutputFileFactory` here, so we do not need to provide the table at every call?

Reply:
Yes, I made this change as part of the update to make refresh more explicit.